### PR TITLE
[Not for main]Audio: Aria: Optimization of Aria

### PR DESCRIPTION
**Pull request by andrula-song**
Optimize the Aria, test on tgl device, the bypass mode can reduce about 51.1% cycles and the normal mode can reduce about 24.0% cycles.

### CODE REVIEW

File: src/audio/aria/aria_generic.c [hunk 1]
```diff
@@ -7,50 +7,93 @@
...
+	int32_t *src = audio_stream_get_rptr(source);
+	int i, n;
 
-	for (size_t i = 0; i < src_size; ++i)
-		max_data = MAX(max_data, ABS(data[i]));
+	while (samples) {
+		n = audio_stream_samples_without_wrap_s32(source, src);
+		n = MIN(samples, n);
+		for (i = 0; i < n; i++)
+			max_data = MAX(max_data, ABS(src[i]));
```
Comment (singalsu) [inline src/audio/aria/aria_generic.c:37]: I'm not sure but is there risk for ABS() to be evaluated twice due to implementation of MAX() macro?

File: src/audio/aria/aria_generic.c [hunk 2]
```diff
@@ -7,50 +7,93 @@
...
+			}
+			gain += step;
+			out += ch_n;
 		}
+		samples -= n;
+		in = cir_buf_wrap(in, cd->data_addr, cd->data_end);
+		out = audio_stream_wrap(sink, out);
 	}
-	cd->gain_state = (cd->gain_state + 1) % ARIA_MAX_GAIN_STATES;
+	cd->gain_state = INDEX_TAB[cd->gain_state + 1];
```
Comment (singalsu) [inline src/audio/aria/aria_generic.c:92]: Nice!

**Review by singalsu — APPROVED** [review]

File: src/audio/aria/aria_generic.c (hunk 1 above)
Comment (andrula-song) [reply src/audio/aria/aria_generic.c]: from the xtensa test result, it is evaluated twice, thanks!

**Comment by andrula-song** [comment]
the failed CI cases are unrelated issues and failed QB cases is fw load failed, unrelated too. Merge it.

**Pull request merged**